pkg/handlers/internalapi/move_dates.go

```diff
@@ -76,29 +77,64 @@ func calculateMoveDates(db *pop.Connection, planner route.Planner, moveID uuid.U
 	return summary, nil
 }
 
+// calculateMoveDatesFromShipment takes stored values on the shipment to calculate the most up-to-date move date ranges
+// this is used to display date ranges for the SM HHG review page and the status timeline on the post-hhg-submission landing page
 func calculateMoveDatesFromShipment(shipment *models.Shipment) (MoveDatesSummary, error) {
 	usCalendar := handlers.NewUSCalendar()
 
+	if shipment.OriginalPackDate == nil {
+		return MoveDatesSummary{}, errors.New("Shipment must have a OriginalPackDate")
+	}
+
+	var mostCurrentPackDate time.Time
+	if shipment.ActualPackDate != nil {
+		mostCurrentPackDate = *shipment.ActualPackDate
+	} else if shipment.PmSurveyPlannedPackDate != nil {
+		mostCurrentPackDate = *shipment.PmSurveyPlannedPackDate
+	} else {
+		mostCurrentPackDate = *shipment.OriginalPackDate
+	}
+
 	if shipment.RequestedPickupDate == nil {
 		return MoveDatesSummary{}, errors.New("Shipment must have a RequestedPickupDate")
 	}
-	lastPossiblePackDay := time.Time(*shipment.RequestedPickupDate).AddDate(0, 0, -1)
-
-	if shipment.EstimatedPackDays == nil {
-		return MoveDatesSummary{}, errors.New("Shipment must have a EstimatedPackDays")
+	var mostCurrentPickupDate time.Time
+	if shipment.ActualPickupDate != nil {
+		mostCurrentPickupDate = *shipment.ActualPickupDate
+	} else if shipment.PmSurveyPlannedPickupDate != nil {
+		mostCurrentPickupDate = *shipment.PmSurveyPlannedPickupDate
+	} else {
+		mostCurrentPickupDate = *shipment.RequestedPickupDate
+	}
+	if shipment.OriginalDeliveryDate == nil {
+		return MoveDatesSummary{}, errors.New("Shipment must have a OriginalDeliveryDate")
+	}
+	var mostCurrentDeliveryDate time.Time
+	if shipment.ActualDeliveryDate != nil {
+		mostCurrentDeliveryDate = *shipment.ActualDeliveryDate
+	} else if shipment.PmSurveyPlannedDeliveryDate != nil {
+		mostCurrentDeliveryDate = *shipment.PmSurveyPlannedDeliveryDate
+	} else {
+		mostCurrentDeliveryDate = *shipment.OriginalDeliveryDate
```

Comment:
I think you might consider

```suggestion
} else if shipment.OriginalDeliveryDate != nil {
		mostCurrentDeliveryDate = *shipment.OriginalDeliveryDate
} else if shipment.EstimatedTransitDays != nil && shipment.RequestedPickupDate != nil {
		mostCurrentDeliveryDate = *shipment.RequestedPickupDate + *shipment.EstimatedTransitDays
}
```

Reply:
I changed it to use the estimated delivery days (also the estimated pack days) to calcluate.  Wrote it a bit differently because of the need to account for holidays/weekends.